frontend/src/components/ExternalResourceModal.js

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button, Modal, Alert, useModal, connectModal,
+} from '@trussworks/react-uswds';
+
+import { isValidURL } from '../utils';
+
+const ESCAPE_KEY_CODE = 27;
+
+const ExternalResourceModal = ({ onOpen, onClose }) => (
```

Comment:
For some reason I couldn't get the modal to disappear if the user clicks outside of it. I looked at how context menu was doing it but no success, but escape does work